.github/workflows/ui-tests.yml

```diff
@@ -34,14 +34,72 @@ jobs:
 
       # GitHub Actions sometimes fail to launch the UI tests. Therefore we retry
       - name: Run Fastlane
-        run: for i in {1..2}; do fastlane ui_tests_${{matrix.target}} && break ; done
-        shell: sh
+        id: run-fastlane
+        run: fastlane ui_tests_${{matrix.target}}
+        continue-on-error: true
+      
+      - name: Recovery - Check if the test runner never began executing
+        id: check-if-test-should-be-retried
+        if: ${{ steps.run-fastlane.outcome == 'failure' }}
+        uses: actions/github-script@v7
+        with:
+          script: |
```

Comment:
`m`: Once, we know this works appropriately, it would be great if we somehow could reduce the duplication.

Reply:
Yes, I agree that this is a lot of duplication